src/main/java/com/google/sps/data/Recipe.java

```diff
@@ -37,22 +40,21 @@ public Recipe(long id, String name, String imageBlobKey, String description, Arr
     this.ingredients = ingredients;
     this.steps = steps;
     this.popularity = popularity;
+    publicRecipeID = 0;
+    published = true;
```

Comment:
Setting this default to be different than your other constructor seems error prone. There's no actual enforcement that this constructor is going to be used _only_ for the case you intend, so it could be misused in the future. Can you use the constructor below, and pass in those values when you need them?

Also, side note: if the recipe is published, the publicRecipeID should never be 0, right?

Reply:
if it is a PublicRecipe, then yes bc it helps me tell between the public/private recipe entities
I could make a recipe type variable? would that be better?